tests/feed.rs

```diff
@@ -136,6 +136,67 @@ fn put() {
     b.put(4, None, proof).unwrap();
 }
 
+#[test]
+/// Put data from one feed into another, while veryfing hashes.
+/// I.e. manual replication between two feeds.
+fn put_with_data() {
+    // Create a writable feed.
+    let mut a = create_feed(50).unwrap();
+
+    // Create a second feed with the first feed's key.
+    let (public, secret) = copy_keys(&a);
+    let storage = Storage::new_memory().unwrap();
+    let mut b = Feed::builder(public, storage)
+        .secret_key(secret)
+        .build()
+        .unwrap();
+
+    // Append 4 blocks of data to the writable feed.
+    a.append(b"hi").unwrap();
+    a.append(b"ola").unwrap();
+    a.append(b"ahoj").unwrap();
+    a.append(b"salut").unwrap();
+
+    for i in 0..4 {
+        let a_proof = a.proof(i, true).unwrap();
+        let a_data = a.get(i).unwrap();
+        eprintln!("A: idx {} data {:?}", i, &a_data);
+        eprintln!("Proof: {:#?}", fmt_proof(&a_proof));
+
+        // When putting with data right away (line after next) without putting
+        // with data = None first, I get errors about "missing tree roots needed for verify".
```

Comment:
@Frando 

i tried to figure out why it doesnt work unless you put None first
i reimplemented your failing test with js version of hypercore like this:
https://gist.github.com/khodzha/36eb72671a087e5e3ae2e0e8b03d55f8

and noticed that for the first `hi` node:
- proof in js has 2 nodes with indices `[2, 5]`
- proof in rust has 3 nodes, indices `[0, 2, 5]`

Reply:
oh, this happens because of `let a_proof = a.proof(i, true).unwrap();`
without `true` proof has 2 nodes and is inserted without problem (although with wrong data :grimacing: )

btw if i do `proof(i, {hash: true})` in js hypercore it will err with `Error: Missing tree roots needed for verify` too